### DISCUSSION TITLE
Where is the "temporary" folder on MacOS?

### DISCUSSION THREAD

**Discussion by mayurjobanputra** [aonez/Keka#1469] [General]
I sometimes use encrypted files to protect sensitive information. I don't want Keka to use a "temp" folder somewhere on my system to hold files during extraction. Is there a way to configure or understand which temp folder is being used (if any)?

**Comment by aonez** [comment]
Hi @mayurjobanputra. Keka creates a temporary `.kextraction` package where the final extraction data will be placed. You can see it if you extract a big file or in a very slow destination (some USB devices). So there’s no other directories involved in the extraction.

if for example you extract `yourfiles.zip` in the desktop, Keka will create `yourfiles.kextraction` in the desktop and move it if succeed or permanently delete if there’s any error.

**Reply by mayurjobanputra** [reply]
Is the .kextraction package always located in the same location as the destination? Are there any keka settings where I can change/override this functionality?

**Reply by aonez** [reply]
> Is the .kextraction package always located in the same location as the destination?

Yes.

> Are there any keka settings where I can change/override this functionality?

No at this moment, but this will be a nice feature to add.

**Reply by aonez** [reply]
I've added a ticked here https://github.com/aonez/Keka/issues/1472 so this does not get lost.
